ocsci/pytest_customization/ocscilib.py

```diff
@@ -70,6 +82,32 @@ def pytest_configure(config):
     """
     if not config.getoption("--help"):
         process_cluster_cli_params(config)
+        # Add OCS related versions to the html report and remove extraneous metadata
+        if not ocsci_config.RUN['cli_params']['deploy']:
+            print("Collecting Cluster versions")
+            # remove extraneous metadata
+            del config._metadata['Python']
+            del config._metadata['Packages']
+            del config._metadata['Plugins']
+            del config._metadata['Platform']
+
+            # add cluster version
+            clusterversion = get_cluster_version()
+            config._metadata['Cluster Version'] = clusterversion
+
+            # add ceph version
+            ceph_version = get_ceph_version()
+            config._metadata['Ceph Version'] = ceph_version
+
+            # add rook version
+            rook_version = get_rook_version()
+            config._metadata['Rook Version'] = rook_version
+
+            # add csi versions
+            csi_versions = get_csi_versions()
+            config._metadata['csi-provisioner'] = csi_versions['csi-provisioner']
+            config._metadata['cephfsplugin'] = csi_versions['cephfsplugin']
+            config._metadata['rbdplugin'] = csi_versions['rbdplugin']
```

Comment:
Please handle corner cases and possible failures in all the `get_*_versions()` functions. For example, for me it is failing like this:
```
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/jenkins/ocs-ci/venv/lib/python3.6/site-packages/_pytest/main.py", line 202, in wrap_session
INTERNALERROR>     config._do_configure()
INTERNALERROR>   File "/home/jenkins/ocs-ci/venv/lib/python3.6/site-packages/_pytest/config/__init__.py", line 671, in _do_configure
INTERNALERROR>     self.hook.pytest_configure.call_historic(kwargs=dict(config=self))
INTERNALERROR>   File "/home/jenkins/ocs-ci/venv/lib/python3.6/site-packages/pluggy/hooks.py", line 311, in call_historic
INTERNALERROR>     res = self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/home/jenkins/ocs-ci/venv/lib/python3.6/site-packages/pluggy/manager.py", line 87, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/home/jenkins/ocs-ci/venv/lib/python3.6/site-packages/pluggy/manager.py", line 81, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/home/jenkins/ocs-ci/venv/lib/python3.6/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/home/jenkins/ocs-ci/venv/lib/python3.6/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/home/jenkins/ocs-ci/venv/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/jenkins/ocs-ci/ocsci/pytest_customization/ocscilib.py", line 109, in pytest_configure
INTERNALERROR>     config._metadata['cephfsplugin'] = csi_versions['cephfsplugin']
INTERNALERROR> KeyError: 'cephfsplugin'
```

Reply:
Addressed in new commit 0b3320ae944cba50c7e83fcdc05776d769ab3df2